packages/lib/src/interactions/InteractionsProvider.tsx

```diff
@@ -36,29 +37,63 @@ function InteractionsProvider(props: { children: ReactNode }) {
   );
 
   const shouldInteract = useCallback(
-    (id: string, event: MouseEvent) => {
-      const registeredKeys = [...interactionMap.values()]
-        .map((params: Interaction) => params.modifierKey)
-        .filter(isDefined);
+    (interactionId: string, event: MouseEvent | WheelEvent) => {
+      const registeredInteractions = [...interactionMap.entries()].map(
+        ([id, interaction]) => ({
+          id,
+          ...interaction,
+        })
+      );
+
+      function isButtonPressed(button?: MouseButton) {
+        if (event instanceof WheelEvent) {
+          return button === undefined;
+        }
```

Comment:
Due to the change of algorithm below and because `WheelEvent` have a `button` equal to 0, we need a way to distinguish `WheelEvent` from `MouseEvent`. 

Otherwise, we have multiple matching interactions (with one for wheel and the other for click) that our algorithm cannot distinguish.

Reply:
> when event is a WheelEvent, can button ever be defined?

Hm, yes in fact when checking for matching interactions in a `WheelEvent`. With `event instanceof WheelEvent || event.button === button`, it means that every interaction is matching for any `WheelEvent`.

I think we have to treat the `Wheel` case more explicitely by having a dedicated value. I tried to add `'Wheel'` instead of `undefined` for `WheelEvent` (so that `button` is always defined for any interaction).

Reply:
Great call! So can `button` ever be `undefined` now?

Reply:
No ! It is always defined in `InteractionEntry`. Props of interaction components such as `Pan` allow `button` to be `undefined` but provide a fallback

Reply:
Oh so you can remove the `?` in the function signature. Sorry, should have been clearer.